### PR TITLE
Changed the way we look for indentation in OnlyTabIdentationInXmlFilesCheck

### PR DESCRIPTION
**Pull request by kssimova**
Fix for #250 
Signed-off-by: Kristina S. Simova <kssimovaa@gmail.com>

### CODE REVIEW

**Comment by kssimova** [comment]
@martinvw It would be great if you take a look. Thanks!

File: custom-checks/checkstyle/src/test/java/org/openhab/tools/analysis/checkstyle/test/OnlyTabIdentationInXmlFilesCheckTest.java [hunk 1]
```diff
@@ -19,56 +19,84 @@
...
+
+    @Test
+    public void testBindingXml() throws Exception {
+        String fileName = "binding";
+        verifyXmlTabIdentation(fileName, noMessagesExpected(), false);
+    }
+
+    @Test
+    public void testThingTypesXml() throws Exception {
+        String fileName = "thing-types";
```
Comment (martinvw) [inline custom-checks/checkstyle/src/test/java/org/openhab/tools/analysis/checkstyle/test/OnlyTabIdentationInXmlFilesCheckTest.java]: Consider giving the files and tests names to express what you are actually trying to test and maybe also isolate test cases

File: custom-checks/checkstyle/src/main/java/org/openhab/tools/analysis/checkstyle/OnlyTabIdentationInXmlFilesCheck.java [hunk 2]
```diff
@@ -11,44 +11,47 @@
...
- * 
+ *
  * @author Lyubomir Papazov - initial contribution
+ * @author Kristina Simova - Changed regular expression to match lines starting with whitespaces
  *
  */
 public class OnlyTabIdentationInXmlFilesCheck extends AbstractStaticCheck {
 
-    private static final String PATTERN_TO_BE_FOLLOWED = "^\\t*[<].*";
+    private static final Pattern PATTERN_TO_BE_FOLLOWED = CommonUtils.createPattern("[^\\S\\n\\t]+[<]");
```
Comment (martinvw) [inline custom-checks/checkstyle/src/main/java/org/openhab/tools/analysis/checkstyle/OnlyTabIdentationInXmlFilesCheck.java]: Personally I see no added value for using this util here?

Comment (martinvw) [reply custom-checks/checkstyle/src/main/java/org/openhab/tools/analysis/checkstyle/OnlyTabIdentationInXmlFilesCheck.java]: Can you also explain what you think that it does

Comment (martinvw) [reply custom-checks/checkstyle/src/main/java/org/openhab/tools/analysis/checkstyle/OnlyTabIdentationInXmlFilesCheck.java]: So if an empty line contains spaces its not reported
If a line does not have any < but does start with spaces it is not reported and more

What is our goal?

Imho we just should check whether a line starts with spaces? Or maybe contains any spaces before the first non whitespace character.

What about the following:

`^\s* +\s*\S`

Comment (kssimova) [reply custom-checks/checkstyle/src/main/java/org/openhab/tools/analysis/checkstyle/OnlyTabIdentationInXmlFilesCheck.java]: You are absolutely right, I will fix it right away!

File: custom-checks/checkstyle/src/main/java/org/openhab/tools/analysis/checkstyle/OnlyTabIdentationInXmlFilesCheck.java [hunk 3]
```diff
@@ -12,48 +12,81 @@
...
+            // if line is empty and does not contain only tabs for indentation
+            if (line.trim().isEmpty() && !doesLineContainOnlyTabs(line)) {
                 if (onlyShowFirstWarning) {
+                    logMessage(lineNumber);
                     return;
                 }
+                logMessage(lineNumber);
+                continue;
+            }
+            int indexNonWhitespaceCharacter = line.indexOf(line.trim());
```
Comment (martinvw) [inline custom-checks/checkstyle/src/main/java/org/openhab/tools/analysis/checkstyle/OnlyTabIdentationInXmlFilesCheck.java]: 👍

**Pull request merged**